etcdctl/ctlv3/command/global.go

```diff
@@ -169,6 +169,23 @@ func newClientCfg(endpoints []string, dialTimeout, keepAliveTime, keepAliveTimeo
 	var cfgtls *transport.TLSInfo
 	tlsinfo := transport.TLSInfo{}
 	tlsinfo.Logger, _ = zap.NewProduction()
+
+	cfg := &clientv3.Config{
```

Comment:
Any reason for this reshuffling? Same argument as on previous case, having refactor combined with tests makes it harder to review. 

Let's just skip this reshuffling as it's not needed in this PR.

Reply:
Actually the implementation of function `newClientCfg` is problematic, because when the input parameter `scfg` is nil, then it will run into panic. The reason why we do not see any issue so far is that the callers guarantee that it will never be nil. 

It doesn't matter whether refactoring this function, but we do need to refactor it after moving it into clientv3.

Anyway, let me revert the change for now. This PR will be only adding unit test.

Reply:
Ok, then it's good that we added test and found the panic. We should also add a test for panic, they can wait after we fix it. As panic is not expected or desired, there is no need to test it.

Reply:
All the test in this PR will be merged into anther PR [13821](https://github.com/etcd-io/etcd/pull/13821). The code refactor and the cases (sub tests) for verifying the panic caused by nil scfg will also be added in that PR.